### PR TITLE
[feat]: [PL-25184]:  Get the latest commit id using a separate api if the get file api request doesn't return it

### PR DESCRIPTION
**Pull request by DeepakPatankar**
You can use the following comments to re-trigger PR Checks

- Compile: `trigger compile`
- runAeriformCheck: `trigger AeriformCheck`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`

---
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/33190)

### CODE REVIEW

**Review by bhavya181 — DISMISSED** [review]
> Reviewed 2 of 2 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/33190)* status: 2 file(s) to review. 1 of 2 approvals obtained.

**Comment by mohitg0795** [comment]
*[`product/ci/scm/file/file.go` line 70 at r1](https://reviewable.io/reviews/harness/harness-core/33190#-N2Ifw2t3hJL2UsaKnRr:-N2Ifw2u4wrJPKzISx_K:b-suzt48) ([raw file](https://github.com/harness/harness-core/blob/4a621b745e811b3eda3c21de91d7ec7686111246/product/ci/scm/file/file.go#L70)):*
> ```Go
>         response, err := git.GetLatestCommitOnFile(ctx, request, log)
>         if err != nil {
>             out = &pb.FileContent{
> ```

I see status is not being returned in this case of error.

**Comment by mohitg0795** [comment]
*[`product/ci/scm/file/file.go` line 71 at r1](https://reviewable.io/reviews/harness/harness-core/33190#-N2Ig25-A7gf_KkpFfJG:-N2Ig25-A7gf_KkpFfJH:bonrn8t) ([raw file](https://github.com/harness/harness-core/blob/4a621b745e811b3eda3c21de91d7ec7686111246/product/ci/scm/file/file.go#L71)):*
> ```Go
>         if err != nil {
>             out = &pb.FileContent{
>                 Error:  err.Error(),
> ```

Also, if this operation fail, we should send some generic error saying "couldn't fetch file content" with a generic error code. It doesn't make sense to send error saying "commit not found" when the operation is a getFileContent operation.

**Review by DeepakPatankar — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/33190)* status: 2 file(s) to review. 2. unresolved discussions. 1 of 2 approvals obtained. 

___
*[`product/ci/scm/file/file.go` line 70 at r1](https://reviewable.io/reviews/harness/harness-core/33190#-N2Ifw2t3hJL2UsaKnRr:-N2LjdgW4eZl8kGxqQ-3:b-u6qhcq) ([raw file](https://github.com/harness/harness-core/blob/4a621b745e811b3eda3c21de91d7ec7686111246/product/ci/scm/file/file.go#L70)):*
<details><summary><i>Previously, mohitg0795 (Mohit Garg) wrote…</i></summary><blockquote>

I see status is not being returned in this case of error.
</blockquote></details>

The GetLatestCommitOnFile Api doesn't return us the status. This required changes adding status in 2 APIs, I have created a separate umbrella ticket to track this https://harness.atlassian.net/browse/PL-25228.
___
*[`product/ci/scm/file/file.go` line 71 at r1](https://reviewable.io/reviews/harness/harness-core/33190#-N2Ig25-A7gf_KkpFfJG:-N2LkMRz06QBxGSy3EQn:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/4a621b745e811b3eda3c21de91d7ec7686111246/product/ci/scm/file/file.go#L71)):*
<details><summary><i>Previously, mohitg0795 (Mohit Garg) wrote…</i></summary><blockquote>

Also, if this operation fail, we should send some generic error saying "couldn't fetch file content" with a generic error code. It doesn't make sense to send error saying "commit not found" when the operation is a getFileContent operation.
</blockquote></details>

Done.

File: product/ci/scm/file/file.go [hunk 1]
```diff
@@ -55,9 +55,30 @@ func FindFile(ctx context.Context, fileRequest *pb.GetFileRequest, log *zap.Suga
 	}
 	log.Infow("Findfile success", "slug", fileRequest.GetSlug(), "path", fileRequest.GetPath(), "ref", ref, "commit id",
 		content.Sha, "blob id", content.BlobID, "elapsed_time_ms", utils.TimeSince(start))
+
+    commitID := content.Sha
```
Comment (shubham149) [inline product/ci/scm/file/file.go:59]: gofmt

Comment (DeepakPatankar) [reply product/ci/scm/file/file.go]: done

**Review by shubham149 — DISMISSED** [review]
> Reviewed 1 of 1 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/33190)* status: 2 file(s) to review. 2. unresolved discussions. 1 of 2 approvals obtained.

**Comment by mohitg0795** [comment]
*[`product/ci/scm/file/file.go` line 70 at r1](https://reviewable.io/reviews/harness/harness-core/33190#-N2Ifw2t3hJL2UsaKnRr:-N2PtMR55PS-huhabwQm:b6fxnor) ([raw file](https://github.com/harness/harness-core/blob/4a621b745e811b3eda3c21de91d7ec7686111246/product/ci/scm/file/file.go#L70)):*
<details><summary><i>Previously, DeepakPatankar (Deepak Patankar) wrote…</i></summary><blockquote>

The GetLatestCommitOnFile Api doesn't return us the status. This required changes adding status in 2 APIs, I have created a separate umbrella ticket to track this https://harness.atlassian.net/browse/PL-25228.
</blockquote></details>

So, we can send hardcoded status over here right? Something like 400 to keep consistency going until we fix the real issue.
Also, anyways we can't use the status from GetLatestCommitOnFile API as it represents something else only. 
As we aren't able to fetch file content eventually, so we can send 400. That should be fine.

**Review by DeepakPatankar — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/33190)* status: 2 file(s) to review. 2. unresolved discussions. 1 of 2 approvals obtained. 

___
*[`product/ci/scm/file/file.go` line 70 at r1](https://reviewable.io/reviews/harness/harness-core/33190#-N2Ifw2t3hJL2UsaKnRr:-N2PveOO2WWXcPHMTnLX:b-t9mgmj) ([raw file](https://github.com/harness/harness-core/blob/4a621b745e811b3eda3c21de91d7ec7686111246/product/ci/scm/file/file.go#L70)):*
<details><summary><i>Previously, mohitg0795 (Mohit Garg) wrote…</i></summary><blockquote>

So, we can send hardcoded status over here right? Something like 400 to keep consistency going until we fix the real issue.
Also, anyways we can't use the status from GetLatestCommitOnFile API as it represents something else only. 
As we aren't able to fetch file content eventually, so we can send 400. That should be fine.
</blockquote></details>

Sending the hardcoded status would be wrong because:

1. The next developer would think the status from the SCM is 400, but actually it is a hard-coded value.
2. GetLatestCommitOnFile represents a part of the API. If the input is wrong or the credentials are wrong then GetLatestCommitOnFile will return the correct error message. 
3. If error handling framework has some predefined meaning of 400 status code, then if I hard code the value then we might end up giving wrong error message

**Comment by mohitg0795** [comment]
*[`product/ci/scm/file/file.go` line 70 at r1](https://reviewable.io/reviews/harness/harness-core/33190#-N2Ifw2t3hJL2UsaKnRr:-N2Q-fuC3s4dAC3wqNzv:bh01qoe) ([raw file](https://github.com/harness/harness-core/blob/4a621b745e811b3eda3c21de91d7ec7686111246/product/ci/scm/file/file.go#L70)):*
<details><summary><i>Previously, DeepakPatankar (Deepak Patankar) wrote…</i></summary><blockquote>

Sending the hardcoded status would be wrong because:

1. The next developer would think the status from the SCM is 400, but actually it is a hard-coded value.
2. GetLatestCommitOnFile represents a part of the API. If the input is wrong or the credentials are wrong then GetLatestCommitOnFile will return the correct error message. 
3. If error handling framework has some predefined meaning of 400 status code, then if I hard code the value then we might end up giving wrong error message
</blockquote></details>

1) Yes, its hardcoded for a purpose as we are intentionally wanting to send HTTP status 400. In case getLatestCommit API fails, we want to indicate that complete operation of getFileContent is failed, as we failed to keep up the contract.
2) If input is wrong or creds are wrong, already the getFileContent API would fail before it.
3) Please ignore error handling framework here. This is an SCM Service API, it doesn't need to think what other services are doing over and above it. It just has to maintain its own correctness.

Let me know if we want to get over a call. Not sending any status code would be a major miss if we do so.

**Review by DeepakPatankar — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/33190)* status: 2 file(s) to review. 2. unresolved discussions. 0 of 2 approvals obtained. 

___
*[`product/ci/scm/file/file.go` line 70 at r1](https://reviewable.io/reviews/harness/harness-core/33190#-N2Ifw2t3hJL2UsaKnRr:-N2R6Jc44t7ZrMt9d36T:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/4a621b745e811b3eda3c21de91d7ec7686111246/product/ci/scm/file/file.go#L70)):*
<details><summary><i>Previously, mohitg0795 (Mohit Garg) wrote…</i></summary><blockquote>

1) Yes, its hardcoded for a purpose as we are intentionally wanting to send HTTP status 400. In case getLatestCommit API fails, we want to indicate that complete operation of getFileContent is failed, as we failed to keep up the contract.
2) If input is wrong or creds are wrong, already the getFileContent API would fail before it.
3) Please ignore error handling framework here. This is an SCM Service API, it doesn't need to think what other services are doing over and above it. It just has to maintain its own correctness.

Let me know if we want to get over a call. Not sending any status code would be a major miss if we do so. 
</blockquote></details>

Done.

**Review by mohitg0795 — APPROVED** [review]
:lgtm:

> Reviewed 1 of 2 files at r1, 1 of 1 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/33190#-:-N2RPYjjCPYwjrDTfewH:bnfp4nl)* status: 1 file(s) to review. 1 of 2 approvals obtained.

**Review by bhavya181 — APPROVED** [review]
> Reviewed 1 of 1 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/33190)* status: :shipit: complete! ✓

**Review by hemanthmantri — COMMENTED** [review]
This PR was merged without `go` label added so it probably skipped go tests?
The //product/ci/scm/file:file_test is failing

**Pull request merged**